### PR TITLE
shields: add generic shield for SSD1306 based 128x32 display

### LINKED ISSUE #13725: drivers: ssd1306: When 128x32 is used, only half of the screen is output.
I have a `128x32` SSD1306 module.
SSD1306 Module(128x32): [Adafruit PiOLED - 128x32 Monochrome OLED Add-on for Raspberry Pi](https://www.adafruit.com/product/3527)

There is height in dts of SSD1306.
``` yaml
height:
  type: int
  category: required
  description: Height in pixel of the panel driven by the controller
  generation: define
```

So I put `32` in the height as shown below.
However, only half of the screen is printed.
```
ssd1306@3c {
        compatible = "solomon,ssd1306fb";
        reg = <0x3c>;
        label = "SSD1306";
        width = <128>;
        height = <32>;
        segment-offset = <0>;
        page-offset = <0>;
        display-offset = <0>;
        segment-remap;
        com-invdir;
        prechargep = <0x22>;
};
```

Below is the output of the screen according to the `height` value.
I tested it with `cfb_shell`.
```
cfb init
cfb print 0 0 1234567890
```

| height = <64> | height = <32> |
| --------------| ------------- |
| image  | image |

I tried to modify it as below.
The results are slightly different, but still half the output.
``` diff
diff --git a/drivers/display/ssd1306.c b/drivers/display/ssd1306.c
index 3c91e3d812..b354d6e4e3 100644
--- a/drivers/display/ssd1306.c
+++ b/drivers/display/ssd1306.c
@@ -32,7 +32,7 @@ LOG_MODULE_REGISTER(ssd1306);
 #define SSD1306_PANEL_NUMOF_PAGES      (DT_SOLOMON_SSD1306FB_0_HEIGHT / 8)
 #define SSD1306_CLOCK_DIV_RATIO                0x0
 #define SSD1306_CLOCK_FREQUENCY                0x8
-#define SSD1306_PANEL_MUX_RATIO                63
+#define SSD1306_PANEL_MUX_RATIO                (DT_SOLOMON_SSD1306FB_0_HEIGHT  - 1)
 #define SSD1306_PANEL_VCOM_DESEL_LEVEL 0x20
 #define SSD1306_PANEL_PUMP_VOLTAGE     SSD1306_SET_PUMP_VOLTAGE_90
``` 
image

Am I using dts incorrectly?
Or is it a driver bug?

### PR DESCRIPTION
**Pull request by jfischer-no**
Add DT properties to support 128x32 display.
Fixes: #13725

Add generic shield for SSD1306 based 128x32 pixel
monochrome displays.

### CODE REVIEW

**Review by vanwinkeljan — APPROVED** [review]

File: boards/shields/ssd1306_128x32/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. _ssd1306_128x32_shield:
+
+SSD1306 128x32 pixels generic shield
+####################################
+
+Overview
+********
+
+Generic shield for OLED displays based on SSD1306 controller with
```
Comment (dbkinder) [inline boards/shields/ssd1306_128x32/doc/index.rst]: Use a complete sentence
```
This is a generic shield for 128x32 pixel resolution OLED displays
based on an SSD1306 controller.
```

File: boards/shields/ssd1306_128x32/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+a resolution of 128x32 pixels. These displays have an I2C interface and
+usually only four pins: GND, VCC, SCL and SDA. Display pins can be
+connected to the pin header of a board using Jump wires.
```
Comment (dbkinder) [inline boards/shields/ssd1306_128x32/doc/index.rst]: using ``jumper`` wires

File: boards/shields/ssd1306_128x32/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Requirements
+************
+
+This shield can only be used with a board which provides a configuration
+for Arduino connectors and defines a node alias for I2C interface
```
Comment (dbkinder) [inline boards/shields/ssd1306_128x32/doc/index.rst]: for ``the`` I2C interface

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by erwango — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for docs, thanks!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**